Updated sound flags for Howl and Heal Bell

AsparagusEduardo opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
`B_UPDATED_MOVE_DATA` wasn't being updated correctly for Howl and Heal Bell

## **Discord contact info**
AsparagusEduardo#6051

LOuroboros commented:
Why rearrange the order of the `.flags` field for those moves though?
It should stay inbetween `.priority` and `.split` imo.

AsparagusEduardo commented:
> Why change the position of the `.flags` field for those moves' data though? Every move should have its data for the `.flags` field after `.priority` in order to remain consistent with Pret's repo imo.

Every instance of `B_UPDATED_MOVE_DATA` has the corresponding checks on top, I'm being consistant with what's currently in BE.

LOuroboros commented:
> Every instance of `B_UPDATED_MOVE_DATA` has the corresponding checks on top

Hmm... okay. Why though?

AsparagusEduardo commented:
> Hmm... okay. Why though?

Otherwise, if we kept the order for them, messy stuff like this would occur:
```c
[MOVE_WHIRLWIND] =
{
    .effect = EFFECT_ROAR,
    .power = 0,
    .type = TYPE_NORMAL,
    #if B_UPDATED_MOVE_DATA >= GEN_6
        .accuracy = 0,
    #else
        .accuracy = 100,
    .pp = 20,
    #endif
    .secondaryEffectChance = 0,
    .target = MOVE_TARGET_SELECTED,
    .priority = -6,
    #if B_UPDATED_MOVE_DATA >= GEN_6
        .flags = FLAG_MAGIC_COAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #elif B_UPDATED_MOVE_DATA == GEN_5
        .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGIC_COAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #else
        .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #endif
    .split = SPLIT_STATUS,
},
```
Instead of just:
```c
[MOVE_WHIRLWIND] =
{
    #if B_UPDATED_MOVE_DATA >= GEN_6
        .accuracy = 0,
        .flags = FLAG_MAGIC_COAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #elif B_UPDATED_MOVE_DATA == GEN_5
        .accuracy = 100,
        .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGIC_COAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #else
        .accuracy = 100,
        .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
    #endif
    .effect = EFFECT_ROAR,
    .power = 0,
    .type = TYPE_NORMAL,
    .pp = 20,
    .secondaryEffectChance = 0,
    .target = MOVE_TARGET_SELECTED,
    .priority = -6,
    .split = SPLIT_STATUS,
},
```
